src/fidasim.f90

```diff
@@ -9640,8 +9627,12 @@ subroutine spectrum(vecp, vi, fields, lambda0, sigma_pi, photons, dlength, lambd
     ! stark-zeeman corrections to energy of n=3 states are -q1, -0.5*q1, 0, 0.5*q1, and q1/2
     q1 = sqrt(4*(e0*h*B/(4*pi*m))**2 + 9*(3*a_0*e0*E)**2)
     ! szratio is gamma/epsilon factor. can be thought of as roughly stark energy shift/ zeeman energy shift.
-    szratio = (e0*h*B/(4*pi*m))/(3*a_0*e0*E)
-
+    if(E.eq.0.d0)then
+    	szratio = 0.d0
+    else
+    	szratio = (e0*h*B/(4*pi*m))/(3*a_0*e0*E)
+    endif
+    allocate(circularity(n_stark))
```

Comment:
Any reason your allocating `circularity` instead of just declaring it like you did earlier?

Reply:
yes, having the array of length 3 for the Lyman-Alpha lines in the else if statement but an array of length 15 in the if statement makes it not work the way I had it before.

Reply:
I'm confused. when you declare `circularity` as `real(Float64), dimension(n_stark) :: circularity` it should be equivalent to `allocate(circularity(n_stark))`